mozilla-services/consvc-shepherd
# Draft: [SHEP-17] - Create campaign overview model and migration

## References

[SHEP-17](https://mozilla-hub.atlassian.net/browse/SHEP-17)

## Problem Statement

As part of the [third milestone of the Ad Dashboard MVP](https://docs.google.com/document/d/1_M9uvN7n0czOju8CQxYFgOjf3VVqGeHCLUffcXLlwgM/edit#heading=h.dgfbcz6spjb), we aim to pull click and impression data from Glean/BQ and import it into Shepherd. This PR represents our first attempt at creating a model, migrations, and an admin view for the DeliveredCampaign overview.

## Proposed Changes

Include a new `DeliveredCampaign` schema in `models.py`
Create migration for `DeliveredCampaign`
Create a new tab on the admin page for the campaign overview.

## Verification Steps

1. Run `make test` and ensure tests pass
1. Run `make dev` to run shepherd locally
2. Exec into the shepherd container and run the migrations
```
python manage.py makemigrations
python manage.py migrate
```
3. Connect to the DB and take a look at the table.  Example with some mock data:
```
docker exec -it consvc-shepherd-db-1 psql -U postgres postgres

postgres=# select * from consvc_shepherd_campaignoverview;
 id | submission_date | flight_id | campaign_id | surface | country | product | provider | clicks | impressions 
----+-----------------+-----------+-------------+---------+---------+---------+----------+--------+-------------
  4 | 2024-09-12      |    123456 |      987654 | mobile  | US      | Firefox | Kevel    |    120 |        1500
(1 row)

```

## Visuals
Admin page:
![image](https://github.com/user-attachments/assets/e539403f-dc67-4385-b3e5-be3902758fe1)

[SHEP-17]: https://mozilla-hub.atlassian.net/browse/SHEP-17?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
